File: src/main/java/com/wafflestudio/draft/dto/request/GetUsersByPreferenceRequest.java
package com.wafflestudio.draft.dto.request;


import lombok.Data;
import org.springframework.format.annotation.DateTimeFormat;

import java.time.DayOfWeek;
import java.time.LocalTime;

@Data
public class GetUsersByPreferenceRequest {
    private String regionName;
    private DayOfWeek dayOfWeek;
    @DateTimeFormat(pattern = "HHmmss")
    private LocalTime startTime;
    @DateTimeFormat(pattern = "HHmmss")
    private LocalTime endTime;
}

File: src/main/kotlin/com/wafflestudio/draft/dto/request/GetUsersByPreferenceRequest.kt
package com.wafflestudio.draft.dto.request

import org.springframework.format.annotation.DateTimeFormat
import java.time.DayOfWeek
import java.time.LocalTime

class GetUsersByPreferenceRequest(
        var regionName: String? = null,
        var dayOfWeek: DayOfWeek? = null,

        @field:DateTimeFormat(pattern = "HHmmss")
        var startTime: LocalTime? = null,

        @field:DateTimeFormat(pattern = "HHmmss")
        var endTime: LocalTime? = null
)
